Added a timeout to state regeneration

rolfyone opened the pull request:
Added a development flag to allow us to set state regeneration timeout, with a 120 second timeout by default, and a tiny bit of sanity around the flag not being less than 1.

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

rolfyone committed:
Added a timeout to state regeneration

Added a development flag to allow us to set state regeneration timeout, with a 120 second timeout by default, and a tiny bit of sanity around the flag not being less than 1.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
spotless

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Merge remote-tracking branch 'upstream/master' into regenerate-state-context

rolfyone committed:
Merge branch 'master' into regenerate-state-context

rolfyone committed:
increased attestation pool size limit (#8110)

32 Slots here in cache still seems appropriate given their relative value beyond that, but we should increase the cache limit to cater for storing more attestations.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Merge remote-tracking branch 'upstream/master' into regenerate-state-context

rolfyone committed:
Merge branch 'master' into regenerate-state-context

rolfyone committed:
Merge branch 'master' into regenerate-state-context

rolfyone committed:
Merge branch 'master' into regenerate-state-context

rolfyone committed:
Merge branch 'master' into regenerate-state-context

tbenr commented on storage/src/main/java/tech/pegasys/teku/storage/server/state/FinalizedStateCache.java:
```diff
@@ -106,13 +136,21 @@ private BeaconState regenerateState(final UInt64 slot, final BeaconState stateFr
+      }
+      final long regenerateSlotCount = slot.minusMinZero(stateFromDisk.getSlot()).longValue();
+      LOG.trace("Slots to regenerate state from: {}", regenerateSlotCount);
+      if (regenerateSlotCount > 10_000L) {
```
we could have this constant statically defined up in the class.

rolfyone replied:
true, potentially at some point configurable, but initially 10k is a decent distance

tbenr reviewed (approved):
LGTM. Yes I checked that the other regeneration is completely separated from this.

rolfyone committed:
broke out the StateCacheLoader class so that a test could be added

Moved the constant in line with review comments but also allowed it to be set for testing.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Merge remote-tracking branch 'upstream/master' into regenerate-state-context

rolfyone committed:
fix from security check

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Merge branch 'master' into regenerate-state-context
